src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -80,9 +128,30 @@ static void create_data(ProgramBuilder& p, const ov::Shape& const_shape, const s
         constTensor.feature[0] = 1;
     }
 
+    cldnn::padding padding_data = cldnn::padding();
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    if (p.get_config().get_use_onednn()
+            && (op->get_element_type() == ov::element::i4 || op->get_element_type() == ov::element::u4)
+            && !op->is_dynamic()
+            && const_shape.size() == 2) {
+        const size_t alignment = 2;
+        if (const_shape.back() % alignment != 0) {
+            std::vector<int32_t> new_paddings(const_shape.size(), 0);
+            ov::Shape new_shape = const_shape;
+            for (size_t i = 0; i < const_shape.size(); ++i) {
+                if (const_shape[i] % alignment != 0) {
+                    new_shape[i] = (const_shape[i] + (alignment-1)) & ~(alignment-1);
+                    new_paddings[i] = new_shape[i] - const_shape[i];
+                }
```

Comment:
We only need to add padding to the innermost dim.

Reply:
But we need to take care of this becuase there might be a transpose not fused at the creation, but to be fused after program pass. 
Therefore we'd better to have a separate pass to apply padding after gpu graph's constant folding.

Reply:
I made ocl kernel instead of CPU computation on constant creation. please review it again. thanks.